src/blockdiagonal.jl

```diff
@@ -102,6 +107,23 @@ end
 
 ## Base
 Base.Matrix(B::BlockDiagonal) = cat(blocks(B)...; dims=(1, 2))
+
+function ChainRulesCore.rrule(::Type{<:Base.Matrix}, B::BlockDiagonal)
+    function Matrix_pullback(Δ::AbstractMatrix)
```

Comment:
What do you think of tightening this to:
```suggestion
function ChainRulesCore.rrule(::Type{<:Base.Matrix}, B::BlockDiagonal)
    function Matrix_pullback(Δ::Matrix)
```

I don't think we need to, and I think we have talked about some operations returning more sparse `AbstractMatrix` for matrix inputs.

Reply:
Seems reasonable to me.